modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -69,6 +72,63 @@ def update_appointment(appt_id, status)
 
       private
 
+      def mobile_facility_service
+        @mobile_facility_service ||=
+          VAOS::V2::MobileFacilityService.new(user)
+      end
+
+      # Entry point for processing appointment responses for converting their times from UTC to local.
+      # Uses the location_id from the appt body to fetch the facility's timezone that is then passed along
+      # with the appointment time to the convert_utc_to_local_time method which does the actual conversion.
+      def convert_appointment_time(appt)
+        if !appt[:start].nil?
+          facility_timezone = get_facility_timezone(appt[:location_id])
+          appt[:start] = convert_utc_to_local_time(appt[:start], facility_timezone)
+        elsif !appt.dig(:requested_periods, 0, :start).nil?
+          appt[:requested_periods].each do |period|
+            facility_timezone = get_facility_timezone(appt[:location_id])
+            period[:start] = convert_utc_to_local_time(period[:start], facility_timezone)
+          end
+        end
+        appt
+      end
+
+      # Returns a local [DateTime] object converted from UTC using the facility's timezone offset.
+      # We'd like to perform this change only on the appointment responses to offer a consistently
+      # formatted local time to our consumers while not changing how we pass DateTimes to upstream services.
+      #
+      # @param [DateTime] date - the date to be modified, required
+      # @param [String] tz - the timezone id, won't convert if nil
+      # @return [DateTime] date in local time, will return in UTC if tz is nil
+      #
+      def convert_utc_to_local_time(date, tz)
+        raise Common::Exceptions::ParameterMissing, 'date' if date.nil?
+
+        date.to_time.utc.in_time_zone(tz).to_datetime
+      end
+
+      FACILITY_ERROR_MSG = 'Error fetching facility details'
+
+      # Returns the facility timezone id (eg. 'America/New_York') associated with facility id (location_id)
+      def get_facility_timezone(facility_location_id)
+        facility_info = get_facility(facility_location_id) unless facility_location_id.nil?
+        if facility_info == FACILITY_ERROR_MSG || facility_info.nil?
+          nil # returns nil if unable to fetch facility info, which will be handled by the timezone conversion
+        else
+          facility_info[:timezone]&.[](:zone_id)
+        end
+      end
+
+      def get_facility(location_id)
+        mobile_facility_service.get_facility_with_cache(location_id)
+      rescue Common::Exceptions::BackendServiceException
```

Comment:
Given that this is happening twice, it seems like it could've happened in the service.